spring-boot-project/spring-boot/src/main/java/org/springframework/boot/context/config/StandardConfigDataLocationResolver.java

```diff
@@ -163,6 +165,21 @@ private Set<StandardConfigDataReference> getProfileSpecificReferences(ConfigData
 		return references;
 	}
 
+	private void validateProfiles(Profiles profiles) {
+		Pattern validProfilePattern = Pattern.compile("^[a-zA-Z0-9_\\-]+$");
```

Comment:
This can be simplified to `Pattern validProfilePattern = Pattern.compile("^[\\w-]+$");`.